pkg/api/kptfile/v1/types.go

```diff
@@ -276,6 +276,24 @@ type Function struct {
 
 	// `ConfigMap` is a convenient way to specify a function config of kind ConfigMap.
 	ConfigMap map[string]string `yaml:"configMap,omitempty"`
+
+	// `Selectors` are used to specify resources on which the function should be executed
+	// if not specified, all resources are selected
+	Selectors []Selector `yaml:"selectors,omitempty"`
+}
+
+// Selector specifies the selection criteria
+type Selector struct {
+	// APIVersion of the target resources
+	APIVersion string `yaml:"apiVersion,omitempty"`
```

Comment:
We should take `Group` and `version` separately. so far I have heard, version won't be very useful.

Reply:
apiVersion field is directly visible in the resource and is simple to understand and users can provide it easily. Empty Groups are confusing, for e.g. Group of Service resource.